code/game/objects/items/weapons/twohanded.dm

```diff
@@ -92,6 +92,11 @@
 	user.put_in_inactive_hand(O)
 	return TRUE
 
+/obj/item/twohanded/mob_can_equip(mob/M, slot) //Unwields twohanded items when they're attempted to be equipped to another slot
+	if(wielded)
```

Comment:
Minor thing, but if you attempt to put a wielded two-handed item into a slot it doesn't belong, it will un-wield it regardless. It might actually be good to add an `&& slot == slot_bitfield_to_slot(slot_flags)` check here to prevent this.

But I'd like to see what other reviewers/maintainers think about this first.

Reply:
Yeah, that pretty much was why I commented it as such on that last revision. The `slot_flags` check works fine for the ones that do go on the back, but not for those that handle it through weight class, like smaller weapons and require manually unwielding them.